### PR TITLE
fix: call `nuxt.ready()` before registering hooks

### PR DESCRIPTION
**Pull request by chz**
@pi0 please check. Fixed issue: https://github.com/nuxt/bridge/issues/918

this happen only 3.9.0. But 3.8.4 work normally

### CODE REVIEW

File: src/utils/dev.ts [hunk 1]
```diff
@@ -207,7 +207,7 @@ class NuxtDevServer extends EventEmitter {
     }
 
     // Remove websocket handlers on close
-    this._currentNuxt.hooks.hookOnce('close', () => {
+    this._currentNuxt.hooks.hook('close', () => {
       this.listener.server.removeAllListeners('upgrade')
```
Comment (danielroe) [inline src/utils/dev.ts:211]:
```suggestion
    const unsubClose = this._currentNuxt.hooks.hook('close', () => {
      unsubClose()
      this.listener.server.removeAllListeners('upgrade')
```

Comment (pi0) [reply src/utils/dev.ts:211]: Is there a chance we might polyfill hookOnce from once place? Ideally in core or bridge but since i support requires more work, in kit loader at least to patch and support `hookOnce`. Making sure bridge keeps working in long term via this patches seems less practical to me.

Comment (danielroe) [reply src/utils/dev.ts:211]: That would be better, for sure.

Comment (pi0) [reply src/utils/dev.ts:211]: Separate topic how long we should support nuxt 2 / bridge for nuxi we can chat later -- specially as first class (which I'm not sure is also being tested against nuxt/nuxt/kit)

In the meantime yeah let's find a better way to make sure at least regressions with hookOnce won't happen in a safer way.

Comment (danielroe) [reply src/utils/dev.ts:211]: we do test bridge against nuxt with ecosystem-ci before every release of nuxt

File: src/utils/dev.ts [hunk 2]
```diff
@@ -207,7 +207,9 @@ class NuxtDevServer extends EventEmitter {
     }
 
     // Remove websocket handlers on close
-    this._currentNuxt.hooks.hookOnce('close', () => {
+    // nuxt bridge doesn't have support for `hookOnce` here
+    const unsubClose = this._currentNuxt.hooks.hook('close', () => {
+      unsubClose()
```
Comment (danielroe) [inline src/utils/dev.ts:212]: With https://github.com/nuxt/bridge/pull/921, I think we can revert this change and just keep the reordered `ready()` call.

```suggestion
    this._currentNuxt.hooks.hookOnce('close', () => {
```

File: src/utils/dev.ts [hunk 3]
```diff
@@ -184,7 +184,7 @@ class NuxtDevServer extends EventEmitter {
         ...this.options.overrides,
       },
     })
-
+    await this._currentNuxt.ready()
```
Comment (danielroe) [inline src/utils/dev.ts:187]:
```suggestion
    await this._currentNuxt.ready()

```
